### ISSUE TITLE
SSL certificate verification

### ISSUE THREAD

**Issue by Forbzy** [influxdata/influxdb-plugin-fluent#22]
Can we bring back the ability to switch off SSL certificate verification? In the previous version of this plugin you could do that.  I'm using influxdb 2.0.4.

**Comment by bednar** [comment]
Hi @Forbzy,

thanks for using our FluentD plugin.

You could switch off SSL verification by: `use_ssl false`. For more info see docs - https://github.com/influxdata/influxdb-plugin-fluent#configuration. Your configuration could looks like:

```xml
<match httpd.access>
  @type copy
  <store>
    @type influxdb2
    url http://influxdb_v2:8086
    token my-token
    bucket my-bucket
    org my-org


    use_ssl false


    time_precision s
    tag_keys ["method", "host", "path"]
    <buffer tag>
      @type memory
      flush_interval 5
    </buffer>
  </store>
</match>
```

Regards

**Comment by bednar** [comment]
@Forbzy Did you mean settings for https://www.rubydoc.info/stdlib/net/Net%2FHTTP:verify_mode?

**Comment by Forbzy** [comment]
Thats not quite what I'm looking for. Setting it to false will disable secure traffic all together. I want to use SSL/TLS but I don't want the certificate verification.

Kind Regards

James

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by bednar** [comment]
@Forbzy thanks for clarification.

I will prepare an option to configure certificate verification... something like:

| Option | Description | Type | Default |
|---|---|---|---|
| verify_mode | Sets the flags for the certification verification at beginning of SSL/TLS session. | OpenSSL::SSL::VERIFY_NONE or OpenSSL::SSL::VERIFY_PEER | OpenSSL::SSL::VERIFY_PEER |

**Comment by Forbzy** [comment]
That would be perfect. Thank you Jakub.

Kind Regards

James
________________________________

**Comment by bednar** [comment]
@Forbzy, issue is fixed in [Milestone 1.7.0](https://github.com/influxdata/influxdb-plugin-fluent/pull/23)

You could use the preview version of plugin by:

```console
fluent-gem install fluent-plugin-influxdb-v2 -v 1.7.0.pre.935
```

and corresponding `FluentD` settings:

```
<match influxdb2.**>
    @type influxdb2

    # InfluxDB URL to connect to (ex. https://localhost:8086).
    url                     https://localhost:8086
    # Access Token used for authenticating/authorizing the InfluxDB request sent by client.
    token                   my-token
    # Turn on/off SSL for HTTP communication.
    use_ssl                 true

    # Sets the flags for the certification verification at beginning of SSL/TLS session.
    # For more info see - https://docs.ruby-lang.org/en/3.0.0/Net/HTTP.html#verify_mode.
    verify_mode             "#{OpenSSL::SSL::VERIFY_NONE}"

    ...
</match>
```
